src/main/scala/com/databricks/sparkdl/DeepImageFeaturizer.scala

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2017 Databricks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.sparkdl
+
+import java.nio.file.{Files, Paths}
+
+import org.apache.spark.image.ImageSchema
+import org.apache.spark.ml.Transformer
+import org.apache.spark.ml.linalg.SQLDataTypes.VectorType
+import org.apache.spark.ml.linalg.Vectors
+import org.apache.spark.ml.param.{Param, ParamMap}
+import org.apache.spark.ml.util.{DefaultParamsWritable, Identifiable}
+import org.apache.spark.sql.{DataFrame, Dataset, Row}
+import org.apache.spark.sql.functions.{col, udf}
+import org.apache.spark.sql.types.StructType
+
+import org.tensorflow.framework.GraphDef
+import org.tensorframes.{Shape, ShapeDescription}
+import org.tensorframes.impl.{DebugRowOps, SerializedGraph, TensorFlowOps}
+
+
+class DeepImageFeaturizer(override val uid: String) extends Transformer with DefaultParamsWritable {
+
+  def this() = this(Identifiable.randomUID("deepImageFeaturizer"))
+
+  final val inputCol: Param[String] = new Param[String](this, "inputCol", "input column name")
+  final val outputCol: Param[String] = new Param[String](this, "outputCol", "output column name")
+
+  // TODO: make modelName a param
+  val modelName = "_test"
```

Comment:
do we need this to be public?

Reply:
this is just a placeholder for the param. I've fixed it #2 .